### ISSUE TITLE
Flow Annotations causing Jest Test to fail.

### ISSUE THREAD

**Issue by cvongrim** [urbanairship/react-native-module#168]
# Preliminary Info

### What Urban Airship dependencies are you using?
urbanairship-react-native: 2.1.3 
react-native:  0.57.5

### What are the versions of any relevant development tools you are using?
n/a

# Report

### What unexpected behavior are you seeing?
When running Jest, Urban Airship throws an error due to flow type annotations.

```
/node_modules/urbanairship-react-native/js/UACustomEvent.js:8
        _name: string;
             ^

    SyntaxError: Unexpected token :
````
### What is the expected behavior?
To be able to continue running my existing Jest Test.

I assume something needs to be added to strip out the Flow annotations when Jest is ran. I would assume this could be a common issue since Jest comes with React Native as the default Testing library.

### What are the steps to reproduce the unexpected behavior?
Run a Jest Test on any component that imports urbanairship-react-native. 

### Do you have logging for the issue?
No

**Comment by rlepinski** [comment]
@cvongrim https://github.com/facebook/jest/issues/607

Can you try using babel-jest?

**Comment by cvongrim** [comment]
@rlepinski I am using it on my project. I am using Flow in all of my code without issue. I only started running into Jest Flow errors when I import Urban Airship. If I remove the Urban Airship import they go away.

**Comment by rlepinski** [comment]
Ok, I have no idea why our plugin would be at fault but I will take a look. If you could provide a sample project that reproduces the issue that would be extremely helpful.

**Comment by cvongrim** [comment]
Here is an [example project](https://github.com/cvongrim/urbanairship-example). Just pull it down, `yarn install` and try running `yarn test`.  On my actual project, I have gone through the steps of fully setting up Urban Airship to where it does receive Push Notifications successfully. I am going to keep looking for a solution too. Thanks!

**Comment by rlepinski** [comment]
Ok, was able to run tests by adding a module mock https://github.com/facebook/jest/issues/2208

```
jest.mock('urbanairship-react-native', () => {
  return {
  }
})
```

**Comment by rlepinski** [comment]
@cvongrim I think you should be unblocked for now if you add the above. I will look into precompiling the module on publish and possibly providing a default mock for jest.

**Comment by cvongrim** [comment]
Great! Added that and that fixed it. I have been trying to figure this out all day. Thanks for looking into it.

**Comment by nelsonchen5** [comment]
Running into this problem as well, added the above, however when the tests are run, I'll receive a "Cannot read property 'setUserNotificationsEnabled' of undefined

![image](https://user-images.githubusercontent.com/31781643/51418038-0524ea00-1b36-11e9-99e2-3e7f69903742.png)

**Comment by cvongrim** [comment]
@nelsonchen5 This looks to be a different issue than what I had. I didn't have a problem with UrbanAirship. UrbanAirship is imported right? Depending on your test, you might need to mock UrbanAirship. Something like the following. 

`
jest.mock('urbanairship-react-native, () => {
  return {
    setUserNotificationsEnabled: jest.fn(),
  };
});
`

**Comment by nelsonchen5** [comment]
@cvongrim Hi, thanks for the response. I tried mocking it and it doesn't seem to recognize UrbanAirship for some reason

**Comment by rlepinski** [comment]
@nelsonchen5 This issue is unrelated to the original issue.

Please open a new issue with more details. We will probably need a project that reproduces the issue to know your exact issue. Most likely the module is not installed properly (npm install) or you are missing an include somewhere.
